src/SAHB.GraphQLClient/Batching/Internal/GraphQLBatchMerger.cs

```diff
@@ -62,29 +62,22 @@ public IGraphQLQuery<T> AddQuery<T>(params GraphQLQueryArgument[] arguments)
             return new GraphQLBatchQuery<T>(this, identifier);
         }
 
-        public async Task<T> GetValue<T>(string identitifer) 
+        public async Task<T> GetValue<T>(string identifier) 
             where T : class
         {
-            if (!_isExecuted)
-                await Execute().ConfigureAwait(false);
-            
-            if (_result.ContainsErrors)
-            {
-                throw new GraphQLErrorException(query: _executedQuery , errors: _result.Errors);
-            }
-            
-            // Create new JObject
-            JObject deserilizeFrom = new JObject();
+            return await GetDeserializedResult<T>(identifier);
```

Comment:
This does not have to be awaited. Please change to:
`
public Task<T> GetValue<T>(string identifier) 
            where T : class
        {
            return GetDeserializedResult<T>(identifier);
        }
`